test/Events.test.js

```diff
@@ -0,0 +1,326 @@
+const expect = require("chai").expect;
+const EventEmitter = require("events");
+const sinon = require("sinon");
+const { GameState } = require("../entities/GameState");
+const { createPlayer } = require("./testutils");
+const { PropertyService } = require("../entities/PropertyService");
+
+describe("Events", () => {
+  let gameState;
+  let mockUI;
+  let eventBusEmitter;
+  let EVENTS;
+
+  beforeEach(() => {
+    gameState = new GameState();
+    gameState.currentPlayerActions = {
+      END_TURN: {
+        execute: () => true,
+      },
+    };
+    mockUI = {
+      startGame: () => true,
+      startTurn: (player) => false,
+      displayAvailableActions: (actions) => false,
+      prompt: () => true,
+      endTurn: () => true,
+      rollingDice: () => true,
+      rollDiceDisplay: (shouldDisplay) => true,
+      payFineDisplay: (shouldDisplay) => true,
+      endTurnDisplay: (shouldDisplay) => true,
+      diceRollResults: (roll1, roll2) => true,
+      rollNormalDice: () => true,
+      rollJailDice: () => true,
+      caughtSpeeding: () => true,
+      playerMovement: (position) => true,
+      payFine: () => true,
+      passGo: () => true,
+      jail: () => true,
+      unknownAction: () => true
+    };
+    eventBusEmitter = new EventEmitter();
+    EVENTS = require("../entities/Events")(eventBusEmitter, mockUI, gameState);
+    gameState.players = [createPlayer({ name: "player1" })];
+  });
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+  it("constructor should return object with associated event listeners", () => {
+    for (property in EVENTS) {
+      expect(typeof EVENTS[property]).to.equal("function");
+    }
+  });
+
+  describe("startGame", () => {
+    it("should emit start turn event", () => {
+      const startTurnSpy = sinon.spy();
+      eventBusEmitter.on("START_TURN", startTurnSpy);
+      EVENTS.START_GAME();
+      expect(startTurnSpy.callCount).to.equal(1);
+    });
+    it("should make ui call", () => {
+      const uiSpy = sinon.spy();
+      mockUI.startGame = uiSpy;
+      EVENTS.START_GAME();
+      expect(uiSpy.callCount).to.equal(1);
+    });
+  });
+  describe("startTurn", () => {
+    it("should reset speeding counter and turn taken flag", () => {
+      gameState.speedingCounter = 1;
+      gameState.turnTaken = true;
+      EVENTS.START_TURN();
+      expect(gameState.speedingCounter).to.equal(0);
+      expect(gameState.turnTaken).to.be.false;
+    });
+    it("should emit continue turn event", () => {
+      const continueTurnSpy = sinon.spy();
+      eventBusEmitter.on("CONTINUE_TURN", continueTurnSpy);
+      EVENTS.START_TURN();
+      expect(continueTurnSpy.callCount).to.equal(1);
+    });
+    it("should make ui call", () => {
+      const startTurnUISpy = sinon.spy();
+      mockUI.startTurn = startTurnUISpy;
+      EVENTS.START_TURN();
+      expect(startTurnUISpy.callCount).to.equal(1);
+      expect(startTurnUISpy.getCall(0).args).to.deep.equal([
```

Comment:
I think Chai's `.members` assertion/expect(ion? lol) would be better suited here ?
```js
expect(args).to.have.deep.members(gameState.players[0])
```
(^I think that that should work, though it looks like it is checking for membership, it is actually checking the entire set to be equal - there `.not` chain will yield a subset, as well as the `.include` chain: double-check @ https://www.chaijs.com/api/bdd/#method_members)